src/librvnpal/src/posix/virtualmemory.c

```diff
@@ -53,10 +53,12 @@ rvn_protect_range(void *start_address, int64_t size, int32_t flags, int32_t *det
         mprotect_flags |= PROT_WRITE;
     if (flags & MPROTECT_OPTIONS_PROT_EXEC)
         mprotect_flags |= PROT_EXEC;
+#ifndef __APPLE__        
```

Comment:
All of this is not required. And this is exactly why not.
We don't need these options, and exposing them cause us problem